### PR TITLE
Enum support for coalesce directives

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
This PR adds support for enums in the `@coalesce` directive, making its behaviour consistent with `@default`

# Issue
#985

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-986.surge.sh

**Review by tbwiss — APPROVED** [review]
LGTM from a code perspective!

File: packages/graphql/src/schema/validation/scalars.ts [hunk 1]
```diff
@@ -19,19 +19,19 @@
 
 import { GraphQLScalarType, Kind } from "graphql";
 
-export const ScalarType = new GraphQLScalarType({
+export const ScalarOrEnumType = new GraphQLScalarType({
     name: "Scalar",
```
Comment (darrellwarde) [inline packages/graphql/src/schema/validation/scalars.ts:23]: **question:** should we rename this?

**Review by darrellwarde — APPROVED** [review]
Broadly looks good to me, just have one question!

File: packages/graphql/src/schema/validation/scalars.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/src/schema/validation/scalars.ts:23]: Oh, good catch, definitely!

File: docs/modules/ROOT/pages/type-definitions/default-values.adoc [hunk 2]
```diff
@@ -48,7 +48,7 @@ When translating from GraphQL to Cypher, any instances of fields to which this d
 
 [source, graphql, indent=0]
 ----
-"""Int | Float | String | Boolean | ID | DateTime"""
+"""Int | Float | String | Boolean | ID | DateTime | Enum"""
 scalar Scalar
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/type-definitions/default-values.adoc:52]:
```suggestion
scalar ScalarOrEnum
```

**Review by darrellwarde — DISMISSED** [review]
One missing remnant from the rename. 🙂

**Pull request merged**